Packs/ServiceDeskPlus/Integrations/ServiceDeskPlus/ServiceDeskPlus.py

```diff
@@ -0,0 +1,853 @@
+
+    incidents = client.get_requests(params=params).get('requests', [])
+
+    if incidents:
+        count = 0
+        last_run_id = last_run.get('id', '0')
+        last_incident_id = last_run.get('id', '0')
+        cur_time = new_last_run.get('time', 0)
+        incident_creation_time = new_last_run.get('time', 0)
+
+        for incident in incidents:
+            if count >= fetch_limit:
+                break
+            # Prevent fetching twice the same incident
+            if incident.get('id') == last_run_id:
+                continue
+            incident_creation_time = int(incident.get('created_time', {}).get('value'))
+            if incident_creation_time >= cur_time:
+                demisto_incidents.append({
+                    'name': f'{incident.get("subject")} - {incident.get("id")}',
+                    'occurred': timestamp_to_datestring(incident_creation_time),
+                    'rawJSON': json.dumps(incident)
+                })
+                count += 1
+                last_incident_id = incident.get('id')
+
+        if demisto_incidents:
+            new_last_run.update({
+                'time': incident_creation_time,
+                'id': last_incident_id
+            })
+
+    if not demisto_incidents:
+        new_last_run.update({
+            'time': time_to
+        })
+
+    demisto.setLastRun(new_last_run)
+    return demisto_incidents
+
+
+def test_module(client: Client):
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+    If 'Fetches incidents' is checked in the instance configurations, this function checks that the entered parameters
+    are valid.
+
+    Args:
+        client: Service Desk Plus client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    if not client.refresh_token:
+        return_error('Please enter a refresh token (see detailed instruction (?) for more information)')
+    try:
+        client.http_request('GET', 'requests')
+        params: dict = demisto.params()
+
+        if params.get('isFetch'):
+            fetch_time = params.get('fetch_time') if params.get('fetch_time') else '1 day'
+            fetch_status = str(params.get('fetch_status')) if params.get('fetch_status') else 'Open'
+            fetch_filter = str(params.get('fetch_filter')) if params.get('fetch_filter') else ''
+            fetch_limit = int(params.get('fetch_limit', '10')) if params.get('fetch_limit') else 10
+
+            date_format = '%Y-%m-%dT%H:%M:%S'
+            time_from = date_to_timestamp(parse_date_range(fetch_time, date_format=date_format, utc=False)[0])
+            time_to = date_to_timestamp(datetime.now(), date_format=date_format)
+            list_info = create_fetch_list_info(str(time_from), str(time_to), fetch_status, fetch_filter, fetch_limit + 1)
+            params = {
+                'input_data': f'{list_info}'
+            }
```

Comment:
this one needs to be re considered.
the idea is to call the fetch like it is called in every fetch, and the only thing that is done is that the fetch checks if it is a test and know not to get the last run. lets talk about this.

Reply:
Moved the fetch parameters to the client and changed test_module to call fetch_incidents if isFetch